pkg/daemon/update.go

```diff
@@ -2155,24 +2155,26 @@ func (dn *Daemon) reboot(rationale string) error {
 	}
 	dn.logSystem("initiating reboot: %s", rationale)
 
-	rebootCmd := rebootCommand(rationale)
-
 	// reboot, executed async via systemd-run so that the reboot command is executed
 	// in the context of the host asynchronously from us
 	// We're not returning the error from the reboot command as it can be terminated by
 	// the system itself with signal: terminated. We can't catch the subprocess termination signal
 	// either, we just have one for the MCD itself.
+	rebootCmd := rebootCommand(rationale)
 	if err := rebootCmd.Run(); err != nil {
 		dn.logSystem("failed to run reboot: %v", err)
 		mcdRebootErr.Inc()
+		dn.logSystem("reboot command failed, waiting for another reboot request..")
+		return fmt.Errorf("reboot command failed, something is seriously wrong")
 	}
-
-	// wait to be killed via SIGTERM from the kubelet shutting down
+	// wait 5 minutes to see if MCD will be killed via SIGTERM from the kubelet shutting down
```

Comment:
I think we no longer need this timer now. With PR https://github.com/openshift/machine-config-operator/pull/2509 , we no longer explicitly stopping kubelet. Everything is getting handled by `systemctl reboot`  . So ideally if that fails, we should increase the counter and return error.

Reply:
Just updated the PR with our latest findings! (: